src/main/java/picard/vcf/AccumulateVariantCallingMetrics.java

```diff
@@ -101,34 +104,22 @@ protected int doWork() {
                     throw new PicardException(String.format("Expected 1 row in the summary metrics file but saw %d", summaryMetricsFile.getMetrics().size()));
                 }
 
-                // re-calculate internal fields from derived fields and add it to the list of summary metrics
+                // re-calculate internal fields from derived fields and merge it into the main summary metrics
                 final CollectVariantCallingMetrics.VariantCallingSummaryMetrics summaryMetrics = summaryMetricsFile.getMetrics().get(0);
                 summaryMetrics.calculateFromDerivedFields(totalHetDepth);
-                summaries.add(summaryMetrics);
+                CollectVariantCallingMetrics.VariantCallingSummaryMetrics.foldInto(collapsedSummary, Collections.singleton(summaryMetrics));
+                collapsedSummary.calculateDerivedFields();
             } catch (IOException e) {
                 throw new PicardException(String.format("Cannot read from metrics files with prefix %s", inputPrefix));
             }
         }
 
-        // now merge all of the accumulated metrics
-        final Collection<CollectVariantCallingMetrics.VariantCallingDetailMetrics> collapsedDetails = new ArrayList<>();
-        sampleDetailsMap.values().forEach(sampleDetails -> {
-            final CollectVariantCallingMetrics.VariantCallingDetailMetrics collapsed = new CollectVariantCallingMetrics.VariantCallingDetailMetrics();
-            CollectVariantCallingMetrics.VariantCallingDetailMetrics.foldInto(collapsed, sampleDetails);
-            collapsed.calculateDerivedFields();
-            collapsedDetails.add(collapsed);
-        });
-        final CollectVariantCallingMetrics.VariantCallingSummaryMetrics collapsedSummary = new CollectVariantCallingMetrics.VariantCallingSummaryMetrics();
-        CollectVariantCallingMetrics.VariantCallingSummaryMetrics.foldInto(collapsedSummary, summaries);
-        collapsedSummary.calculateDerivedFields();
-
         // prepare and write the finalized merged metrics
         final MetricsFile<CollectVariantCallingMetrics.VariantCallingDetailMetrics, Integer> detail = getMetricsFile();
+        collapsedSampleDetailsMap.values().forEach(detail::addMetric);
+        detail.write(detailOutputFile);
```

Comment:
if you are just moving things around here for aesthetics, could you add a line to separate between the detail and summary metrics?